nixos/modules/virtualisation/hyperv-guest.nix

```diff
@@ -25,40 +28,69 @@ in {
     };
   };
 
-  config = mkIf cfg.enable {
-    boot = {
-      initrd.kernelModules = [
-        "hv_balloon" "hv_netvsc" "hv_storvsc" "hv_utils" "hv_vmbus"
-      ];
+  config = mkIf cfg.enable (mkMerge [
+    {
+      boot = {
+        initrd.kernelModules = [
+          "hv_balloon" "hv_netvsc" "hv_storvsc" "hv_utils" "hv_vmbus"
+        ];
 
-      initrd.availableKernelModules = [ "hyperv_keyboard" ];
+        initrd.availableKernelModules = [ "hyperv_keyboard" ];
 
-      kernelParams = [
-        "video=hyperv_fb:${cfg.videoMode}" "elevator=noop"
-      ];
-    };
+        kernelParams = [
+          "video=hyperv_fb:${cfg.videoMode}" "elevator=noop"
```

Comment:
The video name isn't `hyperv_fb` anymore, but should it even be set with the new driver? I have to use either `Set-VMVideo` on the host or the guest's display settings to configure the resolution; setting it here seems to have no effect.

Reply:
I think this should be changed to `Virtual-1` instead. This is mentioned in https://bugs.launchpad.net/ubuntu/+source/linux/+bug/1986713/comments/3 and https://gist.github.com/robert-claypool/af8670b8082e1df020f21fdf57244cbf

I have tried it with `Virtual-1` and the resolution changes correctly.